internal/report/source.go

```diff
@@ -255,9 +257,33 @@ func assemblyPerSourceLine(objSyms []*objSymbol, rs graph.Nodes, src string, obj
 	var lineno = 0
 	var prevline = 0
 	for _, an := range anodes {
-		if filepath.Base(an.file) == srcBase {
+		// Do not rely solely on the line number produced by Disasm
+		// since it is not what we want in the presence of inlining.
+		//
+		// E.g., suppose we are printing source code for F and this
+		// instruction is from H where F called G called H and both
+		// of those calls were inlined.  We want to use the line
+		// number from F, not from H (which is what Disasm gives us).
+		//
+		// So find the outer-most linenumber in the source file.
+		frames, err := o.file.SourceLine(an.address + o.base)
+		found := false
+		if err == nil {
```

Comment:
Nit: Go style is to limit the scope of variables when possible by putting the assignment within if:

```
found := false
if frames, err := o.file.SourceLine(an.address + o.base); err == nil {
   ...
```

Also, should something be done with the error?